src/components/Modals/NewCourse/SelectedRequirementEditor.vue

```diff
@@ -0,0 +1,162 @@
+<template>
+  <div>
+    <div v-if="relatedRequirements.length > 0">
+      <div class="newCourse-title">This class fulfills the following requirement(s):</div>
+      <div v-if="!editMode" class="newCourse-requirements-container">
+        <div class="newCourse-requirements">
+          {{ relatedRequirements.map(it => it.name).join(', ') }}
+        </div>
+      </div>
+      <div v-else class="newCourse-requirements-edit">
+        <edit-single-requirement
+          v-for="relatedRequirement in relatedRequirements"
+          :key="relatedRequirement.id"
+          :name="relatedRequirement.name"
+          :selected="selectedRequirementIDs.includes(relatedRequirement.id)"
+          :isClickable="true"
+          @on-select="selected => toggleSelectRequirement(relatedRequirement.id, selected)"
+        />
+      </div>
+    </div>
+    <div v-if="potentialRequirements.length > 0">
+      <div class="newCourse-title">
+        This class could potentially fulfill the following requirement(s):
+      </div>
+      <div v-if="!editMode" class="newCourse-requirements-container">
+        <div class="newCourse-name">
+          {{ potentialRequirements.map(it => it.name).join(', ') }}
+        </div>
+      </div>
+      <div v-else class="newCourse-requirements-edit">
+        <edit-single-requirement
+          v-for="potentialRequirement in potentialRequirements"
+          :key="potentialRequirement.id"
+          :name="potentialRequirement.name"
+          :selected="selectedRequirementIDs.includes(potentialRequirement.id)"
+          :isClickable="true"
+          @on-select="selected => toggleSelectRequirement(potentialRequirement.id, selected)"
+        />
+        <requirement-radio-select-button
+          v-for="(choices, index) in radioPotentialRequirements"
+          :key="index"
+          :chosenID="getChosenIDForRadioButton(choices)"
+          :choices="choices"
+          @on-select="selectRequirementFromRadioButtons"
+        />
+      </div>
+    </div>
+    <div v-if="!editMode" class="newCourse-link" @click="toggleEditMode()">
+      {{ editRequirementsText }}
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue';
+import EditSingleRequirement from '@/components/Modals/NewCourse/EditSingleRequirement.vue';
+import RequirementRadioSelectButton from '@/components/Modals/NewCourse/RequirementRadioSelectButton.vue';
+
+export type RequirementWithID = { readonly id: string; readonly name: string };
+
+export default Vue.extend({
+  components: { EditSingleRequirement, RequirementRadioSelectButton },
+  props: {
+    editMode: { type: Boolean, required: true },
```

Comment:
Some note on declaring props in this syntax. I think it's a good idea to do this, since `required` is by default false. Declaring props in this way can help to check we don't forget any props at runtime.